taglib/ape/apetag.cpp

```diff
@@ -421,6 +421,11 @@ void APE::Tag::parse(const ByteVector &data)
     const unsigned int keyLength = nullPos - pos - 8;
     const unsigned int valLegnth = data.toUInt(pos, false);
 
+    if(valLegnth >= data.size() || pos > data.size() - valLegnth) {
```

Comment:
The first term is equivalent to `1 > data.size() - valLegnth` and always true if the second term is true because `pos` starts at `0`.

Reply:
> The first term is equivalent to `1 > data.size() - valLegnth`

I don't think this is correct, as mentioned briefly in 3d97905. Consider the case where `data.size() = 100` and `valLegnth = 101`. It's obvious that the original condition `valLegnth >= data.size()` evaluates to true, however `data.size() - valLegnth = 100 - 101 = UINT_MAX` due to both variables being unsigned types which means that `1 > data.size() - valLegnth` evaluates to false. This is why `valLegnth` must first be validated against `data.size()` before it can be subtracted from it.